upgrading/topics/keycloak/changes-19_0_0.adoc

```diff
@@ -0,0 +1,37 @@
+= Slimming the API of KeycloakSession, deprecation of federation
+
+== What has changed
+
+As part of introducing the Storage.X map storage functionality, several APIs around storage have been consolidated and some have been deprecated.
+Several deprecated classes now exist in the modules `keycloak-model-legacy`, `keycloak-model-legacy-private` and `keycloak-model-legacy-services`.
+Also, several methods on KeycloakSession have been deprecated.
+
+== Why it has changed
+
+The new Storage.X map storage functionality will provide a better abstraction for the service layer and will provide a unified interface to various storages.
+The previous distinction of local storage and federated storage will no longer exist, also the distinction between cached and non-cached storage disappears.
+
+The map storage divides storage in different areas, for example users, group, clients, etc.
+Each of this area can be serviced by a different storage backend.
+With tree storage that is a functionality available within map storage, an administrator can combine several storage backends in a single area, storing different attributes and entities in different stores.
+
+== API Changes
```

Comment:
This section should give also code samples of original code and how this code should be rewritten for each of the paragraphs, and that for both legacy store (using `LegacyStoreManagers`) and for new store (e.g. just using `users()` instead of `userLocalStorage()`).

It should also be stated clearly that using `LegacyStoreManagers` will be only supported for the legacy store and not in the new store (regardless it is obvious from the name).

Reply:
I've added the code examples in each paragraph instead in in its own API block to keep things together.